### ISSUE TITLE
Remove NativeId class

### ISSUE THREAD

**Issue by hategan** [ExaWorks/psi-j-python#2]
The NativeId class is an ABC that's meant to be subclassed such that executors return it as on opaque object from `list()` and can be used in `attach()`. However, it can't be fully opaque since the purpose of `attach()` is to allow different processes to submit/monitor a job. The object cannot be communicated directly between processes without some serialization, which should not be implementation specific.

Long story short, a string may be a better class to represent native ids.

**Comment by hategan** [comment]
This was fixed by PR #5
